### PR TITLE
feat(docs): add fluxdoc formatting documentation

### PR DESCRIPTION
**Pull request by sanderson**
This change adds documentation about inline standard library documentation and fluxdoc-compliant formatting. It also updates the contributing guide to point to the new fluxdoc formatting guide.

### Done checklist
- [x] docs/SPEC.md updated **N/A**
- [x] Test cases written **N/A**

### CODE REVIEW

File: docs/fluxdoc.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+[`/libflux/flux-core/src/bin/fluxdoc.rs`](https://github.com/influxdata/flux/blob/master/libflux/flux-core/src/bin/fluxdoc.rs).
+
+## Syntax and structure
+Each `.flux` package file in `/stdlib` should include comments using the
+following syntax and structure.
+Flux comment lines begin with `//`.
+
+### Use Markdown
+Inline Flux documentation uses Markdown or, to be more specific,
+[GitHub Flavored Markdown (GFM)](https://guides.github.com/features/mastering-markdown/).
```
Comment (nathanielc) [inline docs/fluxdoc.md:20]: Technically we support CommonMark https://spec.commonmark.org/ with support for Github Tables and tasks list. FWIW we are using https://docs.rs/crate/pulldown-cmark/0.8.0 under the hood.

File: docs/fluxdoc.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+`fluxdoc` uses comments in `.flux` package files in `/stdlib` to generate and
+output standard library documentation in JSON format.
+The generated JSON is used to build the public-facing Flux standard library
+documentation and ensure documentation is up-to-date and featur-complete with
+each new Flux release.
+
+**Note:** Packages are required to pass the `fluxdoc` linter before being merged
+into `master` unless listed as an exception in
+[`/libflux/flux-core/src/bin/fluxdoc.rs`](https://github.com/influxdata/flux/blob/master/libflux/flux-core/src/bin/fluxdoc.rs).
```
Comment (nathanielc) [inline docs/fluxdoc.md:11]: Should we not mention the exception list? We intent to completely remove that list once its empty. As this reads now it sort of implies you could add packages to that list which we do not want to encourage.

File: docs/fluxdoc.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+For consistency, use the the following Markdown conventions:
+
+- `#` header syntax
+- `-` character for lists
+- Fenced codeblocks
+
+### Package summary documentation
+Add package summary documentation before the package statement in the `.flux` package file.
+Package summary documentation consists of a **headline**, **description**, and
+**metadata** to be used by clients when consuming the JSON `fluxdoc` output.
```
Comment (nathanielc) [inline docs/fluxdoc.md:30]: Packages can also include examples

**Review by nathanielc — COMMENTED** [review]
This is great thanks for writing this up. I have a few questions below.

File: docs/fluxdoc.md (hunk 2 above)
Comment (sanderson) [reply docs/fluxdoc.md]: Removed.

**Review by nathanielc — APPROVED** [review]

**Pull request merged**